bundles/org.eclipse.swt/Eclipse SWT/win32/org/eclipse/swt/graphics/Image.java

```diff
@@ -132,18 +133,100 @@ public final class Image extends Resource implements Drawable {
 	 */
 	static final int DEFAULT_SCANLINE_PAD = 4;
 
-	private Map<Integer, ImageHandle> zoomLevelToImageHandle = new HashMap<>();
-
 	private List<Consumer<Image>> onDisposeListeners;
 
+	private final ImageHandleManager imageHandleManager = new ImageHandleManager();
+
+	private class ImageHandleManager {
+		private Map<Integer, ImageHandle> zoomLevelToImageHandle = new ConcurrentHashMap<>();
+		private Set<ImageHandle> imageHandles = new HashSet<>();
+
+		ImageHandle getImageHandle(int zoom) {
+			final ImageHandle imageHandleFuture = zoomLevelToImageHandle.get(zoom);
+			if (imageHandleFuture == null) {
+				return null;
+			}
+			return imageHandleFuture;
```

Comment:
There is no future involved here and the value of `zoomLevelToHandle.get(zoom)` could directly be returned, couldn't it?